extensions/wikipedia/src/utils/recents.ts

```diff
@@ -0,0 +1,19 @@
+import { useCachedState } from "@raycast/utils";
+import { Locale, useLanguage } from "./language";
+
+export function useRecentArticles() {
+  const [language] = useLanguage();
+  const [readArticlesMap, setReadArticlesMap] = useCachedState<Partial<Record<Locale, string[]>>>("recentArticles");
+
+  const articles = readArticlesMap?.[language] ?? [];
+
+  return {
+    readArticles: articles,
+    addToReadArticles: (article: string) => {
+      setReadArticlesMap({
+        ...readArticlesMap,
+        [language]: [article, ...articles.filter((a) => a !== article)].slice(0, 20),
+      });
```

Comment:
```suggestion
      setReadArticlesMap((r) => ({
        ...r,
        [language]: [article, ...articles.filter((a) => a !== article)].slice(0, 20),
      }));
```
Use the most recent value of `readArticlesMap`

Reply:
🔥


<img width="1000" alt="wikipedia 2023-07-03 at 16 40 03" src="https://github.com/raycast/extensions/assets/113011003/f62bdeff-c183-4259-aa03-80f75ea3bf81">